services/venice-controller/src/main/java/com/linkedin/venice/controller/server/CreateVersion.java

```diff
@@ -792,7 +803,14 @@ public Route emptyPush(Admin admin) {
         responseObject.setKafkaTopic(version.kafkaTopicName());
         responseObject.setKafkaBootstrapServers(version.getPushStreamSourceAddress());
 
-        admin.writeEndOfPush(clusterName, storeName, versionNumber, true);
+        if (!previousVersions.contains(version)) {
+          LOGGER.info(
+              "Sending SOP and EOP for empty push job: {} for store: {} in cluster: {}",
+              pushJobId,
+              storeName,
+              clusterName);
+          admin.writeEndOfPush(clusterName, storeName, versionNumber, true);
+        }
```

Comment:
Can we add LOG that we skip the request sending and add message in response to make sure that customers know that they need to retry or not.

Reply:
Sure